src/data/resolvers/coinMachine.ts

```diff
@@ -76,5 +77,23 @@ export const coinMachineResolvers = ({
         return null;
       }
     },
+    async coinMachinePeriodTimeRemaining(_, { colonyAddress }) {
+      try {
+        const { networkClient } = colonyManager;
+        const coinMachineClient = await colonyManager.getClient(
+          ClientType.CoinMachineClient,
+          colonyAddress,
+        );
+        const periodLength = await coinMachineClient.getPeriodLength();
+        const blockTime = await getBlockTime(networkClient.provider, 'latest');
+        const timeRemaining = bigNumberify(blockTime).mod(
```

Comment:
I think that the logic is not complete here. How can we now how much time left if we don't take into account the time of the last period change? 
It seems to me that the formula need to be something like:

`periodLength - (latest block time - latest period change block)` 

Or am I missing something?

Reply:
You were right in that something was missing here, the formula would actually be `period length - (latest block timestamp % period length)`. Just asked Alex for clarification on this.

As for how do we know when did the last period change occur, that is what the `latest block timestamp % period length` does, if the result of that is `0`, then that means that the period just began. 

That's at least what I understood from what Alex explained to me.

Reply:
So it still seems to me that something is off in your current code. Your current formula is this:
`const timeRemaining = periodLengthInMs.sub(bigNumberify(blockTime).mod(periodLengthInMs),);`

I don't understand how that formula is calculating time remaining. It doesn't take into account when last period change happened (and in my kitchen logic we need that timestamp).

I think that my testing confirms it. When I installed coin machine and go to the Buy tokens page, I saw about 30 min time remaining although I suppose it should be 60 min left (I enabled with default values and the period length is 60 min).

I don't want to be difficult but I don't want to approve before I understand this. Maybe we can have a call together with Alex to go through that as it's an important point.